src/plus/integrations/providers/gitlab/models.ts

```diff
@@ -129,3 +135,18 @@ export interface GitLabProjectREST {
 		path: string;
 	};
 }
+
+export function fromGitLabMergeRequestProvidersApi(pr: ProviderPullRequest, provider: Integration): PullRequest {
```

Comment:
> 1. Why couldn't it be moved to fromProviderPullRequest?

Because its unknown how it behaves for other integration types, we haven't seen it yet. Probably, when we start doing it for Azure we see that it's all the same and then generalize the code.

> 2. Why can't we move the id exception into fromProviderPullRequest as well?

Same. We know that GitLab requires internal ids to merge. But I'm not sure that it's the same case with Azure or Bitbucket.

Also I think that some of those things should be fixed in the library, and chances they are going to be fixed for different providers at different time. So, when they patch GitLab we remove these hacking substitutions from GitLab, when they fix Azure, we apply it to Azure, etc.

> 3. There is `id` and `nodeId` on a PullRequest object - can we use those?

use for converting?

The existing code has been using the internal ID for usage inside of GitLens: https://github.com/gitkraken/vscode-gitlens/blob/786314ebc47d94dd5aac622f14552960fd4a115b/src/plus/integrations/providers/gitlab/models.ts#L92-L105 (line 104: internal id goes to the `id` field, `iid` in fact is same as `number`)

What comes from the ProvidersAPI
- `id` is global, global id is not accepted. If we put it to the result `id` it can produce conflicts with existing usage.
- `nodeId` is not presented in the input. The output has `nodeId`, the `graphQLId` is substituted there (https://github.com/gitkraken/vscode-gitlens/blob/786314ebc47d94dd5aac622f14552960fd4a115b/src/plus/integrations/providers/models.ts#L784-L791 line 789 goes to `nodeId`)

Reply:
Sounds good. We can leave it as is for now and update down the line if it becomes more of a general case.